### PR TITLE
Ana/fix claim rewards bad proof

### LINKED ISSUE #4005: Kusama Claim Rewards: Bad Proof
Claim Rewards is not working on Kusama. Getting all the time `Invalid Transaction: Bad Proof`

### PR DESCRIPTION
**Pull request by Bitcoinera**
Closes #4005

**Description:**

Just to try stuff. Goal is to have Claim Rewards working. @mariopino what do you think?

This one is working but claiming 0.000001 at a time xD

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by faboweb** [comment]
This would claim one era at a time. to claim all, you would need to do this about 80 times

**Comment by Bitcoinera** [comment]
> This would claim one era at a time. to claim all, you would need to do this about 80 times

xD I know, that is why I called it bad fix. But yeah, @mariopino told me the problem is in the script to calculate these rewards

**Comment by Bitcoinera** [comment]
If I do this

```
    stakerRewards.forEach(reward => {
      reward.nominating.forEach(nomination => {
        nomination = JSON.stringify(nomination)
      })
```

I get 



👀

**Comment by faboweb** [comment]
Can you try using the old payouts before era 718? Or we just show a warning, that you can't claim rewards of eras less than 718 and filter those eras out for claiming.

**Comment by Bitcoinera** [comment]
> Can you try using the old payouts before era 718? Or we just show a warning, that you can't claim rewards of eras less than 718 and filter those eras out for claiming.

Interesting experiment. I'm going to try that

**Comment by Bitcoinera** [comment]
Worked 👌

**Comment by Bitcoinera** [comment]
I'm just wondering, how comes that it worked for `allClaimingTxs[0]`? That one should be the oldest reward... ah I know, the problem was mixing them

File: src/signing/networkMessages/kusama.js [hunk 1]
```diff
@@ -46,11 +46,11 @@ export async function ClaimRewardsTx(senderAddress) {
   let allClaimingTxs = []
   const api = await getAPI()
   const stakerRewards = await api.derive.staking.stakerRewards(senderAddress)
-
-  if (stakerRewards.length === 0) {
+  const newStakerRewards = stakerRewards.filter(({ era }) => era.words[0] > 718)
```
Comment (faboweb) [inline src/signing/networkMessages/kusama.js:49]:
```suggestion
  const newStakerRewards = stakerRewards.toJSON().filter(({ era }) => era > 718)
```

tried this?

**Comment by Bitcoinera** [comment]
Maybe we can then just batch them in two (old and new) for now?

**Comment by faboweb** [comment]
Can we add the warning?

**Comment by Bitcoinera** [comment]
> Can we add the warning?

Sure 👌

**Comment by Bitcoinera** [comment]
Mmm, it seems it is not immeadiate this claim rewards...



But I can see here that it went though: https://polkascan.io/pre/kusama/transaction/0xa2427c82e7b77b5e74e11c3351b2d693f760f3df66b37c9f1d4b6449ae9e42e8

UX challenges

**Comment by Bitcoinera** [comment]
Does it look good?

**Comment by faboweb** [comment]
Maybe with my rewards fix in the API this will be resolved

File: src/signing/networkMessages/kusama.js (hunk 1 above)
Comment (Bitcoinera) [reply src/signing/networkMessages/kusama.js:49]: It doesn't work...

Comment (faboweb) [reply src/signing/networkMessages/kusama.js:49]: true cause it is an array

Comment (faboweb) [reply src/signing/networkMessages/kusama.js:49]:
```suggestion
  const newStakerRewards = stakerRewards.filter((reward) => reward.toJSON().era > 718)
```

**Comment by Bitcoinera** [comment]
> Maybe with my rewards fix in the API this will be resolved

Just tried it and got this again 🤷

**Comment by faboweb** [comment]
So the batch only contained validator_payouts and it still failed?

**Comment by faboweb** [comment]
I successfully claimed on Marios account

**Comment by faboweb** [comment]
Merging. We can improve later.

**Pull request merged**